graphql/server.go

```diff
@@ -344,8 +387,17 @@ func Handler(schema *Schema) http.Handler {
 	})
 }
 
+func (c *conn) Use(fn MiddlewareFunc) {
+	c.middlewares = append(c.middlewares, fn)
+}
+
 func ServeJSONSocket(ctx context.Context, socket JSONSocket, schema *Schema, makeCtx MakeCtxFunc, logger GraphqlLogger) {
```

Comment:
I'd be fine breaking the ServeJSONSocket interface. I am not a big fan of the Use function.